parmed/openmm/parameters.py

```diff
@@ -101,10 +107,20 @@ def from_parameterset(cls, params, copy=False):
         new_params.parametersets = params.parametersets
         new_params._combining_rule = params.combining_rule
         new_params.residues = params.residues
+        # Assign Lennard-Jones atom classes
+        ljtypes = dict()
+        counter = 0
+        for name, atom_type in iteritems(new_params.atom_types):
+            key = (atom_type.rmin, atom_type.epsilon, atom_type.rmin_14,
+                    atom_type.epsilon_14)
+            if key not in ljtypes:
+                ljtypes[key] = counter
```

Comment:
-1

This is an Amber-specific implementation detail, and ParmEd tries to avoid those that don't provide a tangible benefit.  IMO, there is only one reason to define L-J classes that are more general than atom types, and that's to compress the size of the L-J A- and B-coefficient tables so they don't blow GPU shared memory when we have NBFIXes (which some Amber FFs do, and more CHARMM FFs do).  If that was not a consideration, there is absolutely no advantage (at all) for using classes instead of types to define L-J parameters.  It just adds unnecessary complexity in the conversion script without any tangible benefit.

Also, it means the same code is optimal for both CHARMM and AMBER parameter file conversions.  IMO, this is a real win.

So my vote is either 

a) Compress as much as possible the way I'm doing now
b) Don't define classes at all, and just use type for everything

The disadvantages of this proposal:

1) Does not compress types as much as possible, which increases the possibility that NBFIX tables will overflow shared memory.
2) Requires Amber-specific implementation details that are parm.dat-file specific (there is no way to do this assignment in an frcmod file, for instance)

I don't see a real advantage in this scheme except that a tiny number of people (i.e., you, me, and maybe 2 or 3 other people) will see the parallels (and even here, where's the tangible benefit?)